frontend/packages/dev-console/src/components/dropdown/ApplicationDropdown.tsx

```diff
@@ -41,86 +46,73 @@ interface ApplicationDropdownProps {
   onLoad?: (items: { [key: string]: string }) => void;
 }
 
-const ApplicationDropdown: React.FC<ApplicationDropdownProps> = ({ namespace, ...props }) => {
-  const resources = [
-    {
-      isList: true,
-      namespace,
-      kind: 'DeploymentConfig',
-      prop: 'deploymentConfigs',
-    },
-    {
-      isList: true,
-      namespace,
-      kind: 'Deployment',
-      prop: 'deployments',
-    },
-    {
-      isList: true,
-      kind: 'StatefulSet',
-      namespace,
-      prop: 'statefulSets',
-    },
-    {
-      isList: true,
-      kind: 'DaemonSet',
-      namespace,
-      prop: 'daemonSets',
-    },
-    {
-      isList: true,
-      kind: referenceForModel(ServiceModel),
-      namespace,
-      prop: 'knativeService',
-      optional: true,
-    },
-    {
-      isList: true,
-      kind: 'Secret',
-      namespace,
-      prop: 'secrets',
-    },
-    {
-      isList: true,
-      kind: VirtualMachineModel.kind,
-      namespace,
-      prop: 'virtualMachines',
-      optional: true,
-    },
-    {
-      isList: true,
-      kind: CronJobModel.kind,
-      namespace,
-      prop: 'cronjobs',
-      optional: true,
-    },
-    {
-      isList: true,
-      kind: JobModel.kind,
-      namespace,
-      prop: 'jobs',
-      optional: true,
-    },
-    {
-      isList: true,
-      kind: PodModel.kind,
-      namespace,
-      prop: 'pods',
-      optional: true,
-    },
-    ...getDynamicChannelResourceList(namespace),
-    ...getDynamicEventSourcesResourceList(namespace),
-    ...knativeEventingResourcesBroker(namespace),
-  ];
+const applicationSelector = ['metadata', 'labels', 'app.kubernetes.io/part-of'];
+
+const ApplicationDropdown: React.FC<ApplicationDropdownProps & StateProps> = ({
+  namespace,
+  kindsInFlight,
+  ...props
+}) => {
+  const { t } = useTranslation();
+  const [resources, setResources] = React.useState<FirehoseResult[]>([]);
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState();
+  const modelFactories = useExtensions<TopologyDataModelFactory>(isTopologyDataModelFactory);
+
+  const watchedResources = React.useMemo(() => {
+    const watchedBaseResources = getBaseWatchedResources(namespace);
+    return modelFactories.reduce((acc, modelFactory) => {
+      const factoryResources: WatchK8sResources<any> = modelFactory.properties.resources
+        ? modelFactory.properties.resources(namespace)
+        : {};
+      return { ...acc, ...factoryResources };
+    }, watchedBaseResources);
+  }, [modelFactories, namespace]);
+
+  const watchResults = useK8sWatchResources(watchedResources);
+  React.useEffect(() => {
+    const resourcesLoaded =
+      !kindsInFlight &&
+      Object.keys(watchResults).length > 0 &&
+      Object.keys(watchResults).every(
+        (key) => watchResults[key].loaded || watchResults[key].loadError,
```

Comment:
Do you need `kindsInFlight`? Seems like you can rely on `loaded` of the watched resources. `kindsInFlight` will include every resource that's currently in flight from anywhere on the page.

Reply:
Initially loaded can return true with load errors for every resource fetched when `kindsInFlight` is true. This results in an error being shown briefly.

Reply:
I've put a fix into `useK8sWatchResources` to not set the loaded flag to true when the model is not loaded and kinds are still in flight. @rawagner PTAL